cloud_provider/aws/aws_minion_manager.py

```diff
@@ -635,7 +656,28 @@ def check_scaling_group_instances(self, scaling_group):
                 # Wait for sometime before checking again.
                 time.sleep(60)
         return False
-
+    
+    def check_insufficient_capacity(self, scaling_group):
+        """
+        Checks whether not completed ASG activities got not have sufficient capacity error message.
+        """
+        # This error message from https://docs.aws.amazon.com/autoscaling/ec2/userguide/ts-as-capacity.html#ts-as-capacity-1
+        INSUFFICIENT_CAPACITY_MESSAGE = ['We currently do not have sufficient',
+                                           'capacity in the Availability Zone you requested']
+        
+        asg_info = scaling_group.get_asg_info()
+        response = AWSMinionManager.describe_asg_with_retries(
+            self._ac_client, asg_info.AutoScalingGroupName)
```

Comment:
Shouldn't this be `[asg_info.AutoScalingGroupName]`? The second argument to `AWSMinionManager.describe_asg_with_retries` is an array.

Reply:
Sorry, some copy and paste error. And didn't test again after that.
Just tested with real ASG. (still not hit insufficient resource...)